### DISCUSSION TITLE
Curso Dominios Ricos (Dúvida Handlers)

### DISCUSSION THREAD

**Discussion by andregr15** [balta-io/forum#546] [Tira Dúvidas] [answered]
No final do vídeo de handlers, fica claro que para a implementação da função handle para o CreatePayPalSubscriptionCommand é muito semelhante ao CreateBoletoSubscriptionCommand, não seria possível reutilizar essa parte que repete de alguma forma?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @andregr15, como vai?

É possível sim, inclusive criar uma classe base que ambas derivam, a questão é se vale ou não a pena.
Se o cenário é estático, sem mudanças, compensa sim, agora se a ideia é ter comandos mais flexíveis e independentes, aí não compensa!

**Reply by andregr15** [reply]
Entendi, é que como você tinha dito que só mudava o objeto que seria utilizado para o handler executar o comando, pensei em passar o objeto para o handler só fazer a execução que é comum para todos. 
Outra coisa que percebi também foi a repetição do contract, com ele também é possível a reutilização?
Por exemplo, ele ser o mesmo para o Name e criar um NameDTO para o command e reaproveitar o mesmo contract?

**Reply by andrebaltieri** [reply]
Sim! Tudo é possível reaproveitar, só precisa pensar nestes pontos que falei.
Eu vejo muita gente criando tudo genérico e depois especializando o genérico de diversas formas... ou seja, não adianta nada.

Mas se faz sentido para o seu cenário (Ou até mesmo para testes), manda bala 🚀
